src/sagas/account.js

```diff
@@ -37,13 +37,22 @@ export function * doSignup (action) {
 
   try {
     const response = yield minDelayCall(api.signup, formData);
-    response['nextRoute'] = formData.nextRoute;
 
-    // Persist token for future page loads
-    const tokenName = yield select(getTokenName);
-    new Cookies().set(tokenName, response.key, { path: '/' });
+    // If the response had an error status, but wasn't caught as an error
+    if (!!response.status && response.status > 299) {
+      yield put({
+        type: SIGNUP_ERROR, 
+        error: _.get(response, 'data'),
```

Comment:
As stated below, if the `then` statements in `api.js` are left alone, this need not parse out `data`. It would just read:

```javascript
error: response
```